### ISSUE TITLE
Bug - Can't click react-select component in test

### ISSUE THREAD

**Issue by wdfinch** [codesandbox/codesandbox-client#6257]
# 🐛 bug report

## Preflight Checklist

- [x] I have read the [Contributing Guidelines][contributing] for this project.
- [x] I agree to follow the [Code of Conduct][code_of_conduct] that this project
      adheres to.
- [x] I have searched the issue tracker for an issue that matches the one I want
      to file, without success.

## Description of the problem

Hi all, I am having an issue running tests inside of code sandbox. In my test, I want to select an option from a select input. The library I am using to generate the select input is [react-select](https://react-select.com/home). To select an option, I simply want to click the rendered element in my test using [react-testing-library](https://testing-library.com/docs/react-testing-library/intro/). My issue is whenever I try to click the element I get the following error in my test:

`Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'.`

This error does not occur when I run the test outside of code sandbox. I've tried mocking `window.getComputedStyled` in jest but this does not work. 

## How has this issue affected you? What are you trying to accomplish?

I am facing another issue related to react-testing-library with react-select and react-hook-form that I am trying to reproduce in a code sandbox. In order to reproduce my other issue and get help, I need to be able to select an option using react-select. Because I can't seem to do this, I can't resolve my other issue.

### To Reproduce

1. Click on the link below to go to the attached sandbox.
2. Navigate to the test window on the right in the sandbox.
3. Notice that the test fails and the error occurs. 

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Сodesandbox      | https://codesandbox.io/s/frosty-cache-1gybp?file=/src/App.test.js             |
| Browser          |  Chrome: Version 95.0.4638.69 (Official Build) (64-bit) |
| Operating System |  Ubuntu: 20.04 LTS            |

[code_of_conduct]: https://github.com/codesandbox/codesandbox-client/blob/master/CODE_OF_CONDUCT.md
[contributing]: https://github.com/codesandbox/codesandbox-client/blob/master/CONTRIBUTING.md

**Comment by JamesACS** [comment]
Hey @wdfinch 

This seems to be as you've not defined `("Select...")` as an element, if this is removed the following error appears: 

`Unable to find the "window" object for the given node. Please file an issue with the code that's causing you to see this error: https://github.com/testing-library/dom-testing-library/issues/new`

This seems to be a similar issue that can be resolved by ensuring that you're querying the element correctly.

**Comment by wdfinch** [comment]
Hi @JamesACS thanks for the reply. I'm not sure what you mean by the following:

> This seems to be as you've not defined ("Select...") as an element, if this is removed the following error appears:

`<Select/>` is rendered on line 37 in `App.js`. Could you please clarify what you mean or adjust the sandbox to to show how I can view that other error message? I can file an issue with `react-select` if something is not working on their end.

Thanks!

**Comment by JamesACS** [comment]
Sorry @wdfinch - I think I may have removed a line that causing your specific error and generated one of my own 😅

The issue seems to be with this code in app.test.js - If this is removed the components work perfectly. 

`test("selects an option from select input", async () => {
  render(<App />);
  userEvent.click(screen.getByText("Select..."));
});`

So from my understanding, your code needs to scope the test within a window to avoid the error. 

Equally this sort of error can also occur when a dropdown doesn't have dropdown options - so that may also be a solution.

**Comment by wdfinch** [comment]
Thanks for the reply. So the whole point of what I'm trying to do here is run the test. Therefore I don't want to remove that code as it works when run outside of code sandbox. 

> So from my understanding, your code needs to scope the test within a window to avoid the error.

Could you explain how I could do this or modify the sandbox to do this? My goal here is I want the test to click on the `react-select` component without throwing an error. Again this works fine when done outside of code sandbox.

> Equally this sort of error can also occur when a dropdown doesn't have dropdown options - so that may also be a solution.

I am passing in options to the `react-select` component so that shouldn't be the case.

**Comment by wdfinch** [comment]
Got it thanks! Is there a ticket where I can follow that issue or is it internal? Just wanted to know when it's fixed since I have some other unrelated issues that depend on this.

**Comment by JamesACS** [comment]
It's internal, but I'll drop you an update here when I have one. 

Is there any context from your other issues that you think could be useful for us here?

**Comment by wdfinch** [comment]
Not really for this one. I currently facing another bug somewhere in the `jest-dom` / `react-testing-library` space. In order to have someone look at it, I need to reproduce it in a code sandbox, which I can't do because of this bug. So that's how I got here.

**Comment by wdfinch** [comment]
Here's a link to the other bug I'm facing if you're interested. It's unrelated to code sandbox.

https://github.com/testing-library/user-event/issues/777#issuecomment-969399070

**Comment by JamesACS** [comment]
Thank you @wdfinch - I'll add this to the context :)

**Comment by wdfinch** [comment]
Any updates?

**Comment by CompuIves** [comment]
I found the issue. We run the tests in `JSDOM`, because we need to refresh the dom every time a test has finished, to do this, we set the global `window` and `document` when running a file. However, in this case the global `getComputedStyle` is called directly (instead of calling `window.getComputedStyle`), and so the native `getComputedStyle` is run instead of the one in JSDOM. We should ideally change `this` when running the files, but I remember there were some challenges with that. Going to play with that now...
